### PR TITLE
Added Homepage With post Functionlity

### PR DESCRIPTION
**Pull request by rahulcs754**
List of task 
- Added private routes
- Added Home page design
- Added Follow and unfollow functionality
- Remove update Post logged user 
-  Bookmarked function and unbookmarked functionality
 - update user profile pics
 - Update user Profile basic info

### CODE REVIEW

**Review by nakulsharma15 — APPROVED** [review]
Everything Looks good. No changes suggested. I Liked the way you named the functions and the use of modular CSS.

File: src/backend/db/users.js [hunk 1]
```diff
@@ -8,46 +8,81 @@ import { formatDate } from "../utils/authUtils";
...
     lastName: "shukla",
-    username: "rahul@gmail.com",
+    username: "rahul",
+    email: "rahul@gmail.com",
     password: "test123",
+    pic: "",
+    link: "rahulcs754.netlify.com",
+    bio: "Web Developer || Youtuber || Blogger || Software Engineer",
+    followers: [],
+    following: [],
```
Comment (coldpigli) [inline src/backend/db/users.js:36]: Would suggest to keep some mock followers and following users

File: src/frontend/Components/Model/EditModel.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { useRef } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { editPost } from "store/postSlice";
+import styles from "./EditModal.module.css";
+export const EditModel = ({ post, setEdit }) => {
+  const dispatch = useDispatch();
+  const updateInput = useRef();
```
Comment (coldpigli) [inline src/frontend/Components/Model/EditModel.js:7]: Good that you used ref instead of a state here

**Review by coldpigli — APPROVED** [review]
Everything looks good. Have given a few suggestions.

File: src/backend/db/users.js (hunk 1 above)
Comment (rahulcs754) [reply src/backend/db/users.js:36]: okay sure i will do

File: src/frontend/Components/Model/EditModel.js (hunk 2 above)
Comment (rahulcs754) [reply src/frontend/Components/Model/EditModel.js:7]: thanks

**Pull request merged**